### PR TITLE
Refactor code, add Enum deprecation utils

### PR DESCRIPTION
**Pull request by ndonkoHenri**
`ThemeVisualDensity.ADAPTIVEPLATFORMDENSITY` is now deprecated, and `ThemeVisualDensity.ADAPTIVE_PLATFORM_DENSITY` should be used instead.
The deprecation class is in the `theme.py` file.

### CODE REVIEW

File: sdk/python/packages/flet-core/src/flet_core/utils.py [hunk 1]
```diff
@@ -159,3 +161,15 @@ def with_degrees(self, value):
         radians = math.radians(value)
         magnitude = abs(self)
         return type(self).polar(radians, magnitude)
+
+
+class DeprecatedEnumMeta(EnumMeta):
+
```
Comment (FeodorFitsner) [inline sdk/python/packages/flet-core/src/flet_core/utils.py:167]: I would put that class next to `ThemeVisualDensity` (in the same `[src/flet_core/theme.py](https://github.com/flet-dev/flet/pull/2259/files#diff-24599f89970071ace465e406200accb4cdd5a8be6c55fe1a1f1e25618db7516c)` file) and call it `ThemeVisualDensityDeprecated`. What do you think?

Comment (ndonkoHenri) [reply sdk/python/packages/flet-core/src/flet_core/utils.py:167]: Leaving as-is will permit us to use this class on any Enum (add new ifs/cases to the getattribute method), just by passing it as metadata(see VisualDensity). Wdyt?

Comment (FeodorFitsner) [reply sdk/python/packages/flet-core/src/flet_core/utils.py:167]: My point is `DeprecatedEnumMeta` is not targeting specific enum, i.e. you are checking just for `item == 'ADAPTIVEPLATFORMDENSITY'` without checking what enum is that. Imagine the switch will grow up for more obsolete enum members.

I suggest having "Deprecated" sibling for every enum that has deprecated members and keep that class next to its enum.

Comment (ndonkoHenri) [reply sdk/python/packages/flet-core/src/flet_core/utils.py:167]: Ah, i get your point now! :)
Will give it a look.

**Review by FeodorFitsner — APPROVED** [review]

**Pull request merged**